instructionAPI/src/InstructionDecoder-power.C

```diff
@@ -1165,23 +1296,31 @@ which are both 0).
   void InstructionDecoder_power::BO() {
     bcIsConditional = true;
     invertBranchCondition = false;
-    if(!field<8, 8>(insn)) {
-      Expression::Ptr ctr = makeRegisterExpression(makePowerRegID(ppc32::ctr, 0));
+    bool const writes_ctr = (field<8, 8>(insn) == 0);
+
+    if(writes_ctr) {
       if(field<9, 9>(insn)) {
         insn_in_progress->getOperation().mnemonic = "bdz";
       } else {
         insn_in_progress->getOperation().mnemonic = "bdn";
       }
-      insn_in_progress->appendOperand(ctr, true, true);
     }
+
+    // The count register (ctr) is always read, but conditionally written
+    {
+      auto ctr = makeRegisterExpression(makePowerRegID(ppc32::ctr, 0));
+      constexpr bool is_read = true;
+      insn_in_progress->appendOperand(std::move(ctr), is_read, writes_ctr);
+    }
+
     if(!(field<6, 6>(insn))) {
       invertBranchCondition = !field<7, 7>(insn);
       if(insn_in_progress->getOperation().mnemonic == "bc") {
         insn_in_progress->getOperation().mnemonic = "b";
       }
       insn_in_progress->appendOperand(makeBIExpr(), true, false);
     }
-    if(field<8, 8>(insn) && field<6, 6>(insn)) {
+    if(writes_ctr && field<6, 6>(insn)) {
```

Comment:
writes_ctr checks for field<8,8>(insn) ==0, so here it should be (!writes_ctr) I assume?